imports/plugins/included/product-variant/containers/productGridContainer.js

```diff
@@ -38,8 +40,16 @@ const wrapComponent = (Comp) => (
       const selectedProducts = Reaction.getUserPreferences("reaction-product-variant", "selectedGridItems");
       const { products } = this;
 
+      // close the `Grid Settings: Product` panel if the user has not products
       if (_.isEmpty(selectedProducts)) {
-        return Reaction.hideActionView();
+        const actionView = Reaction.getActionView();
+
+        // if some other action view is open, do not hide it
+        if (actionView.template === ACTION_VIEW_TEMPLATE) {
+          Reaction.hideActionView();
```

Comment:
toggling the ActionView causes a re-render of the main section of a page, which is likely where this Component is being rendered, and therefore causes _itself_ to be re-rendered.
can we think of a better way to do this?